File: app/wherebear_app/wherebear_app/Theme/BearTheme.swift
// BearTheme.swift — 設計 tokens（深色底＋暖熊 accent）
// 對應 mockup 色票：bg #17130F / surface #221B13 / cream #F4E3C8 / honey #C98A4B / #E8B878
import SwiftUI

extension Color {
    init(hex: UInt32) {
        self.init(
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255
        )
    }
}

enum BearTheme {
    static let bg          = Color(hex: 0x17130F) // 主背景（暖近黑）
    static let surface     = Color(hex: 0x221B13) // 卡片
    static let surfaceHi   = Color(hex: 0x262019) // modal
    static let sheet       = Color(hex: 0x201913) // bottom sheet
    static let cream       = Color(hex: 0xF4E3C8) // 主文字
    static let honey       = Color(hex: 0xC98A4B) // accent
    static let honeyLight  = Color(hex: 0xE8B878) // accent 亮
    static let ink         = Color(hex: 0x2A1D10) // accent 上的深字
    static let pawInk      = Color(hex: 0x3A2A18) // 掌印
    static let green       = Color(hex: 0x7FAE63) // 回報中
    static let greenText   = Color(hex: 0xCFE3BD)
    static let amber       = Color(hex: 0xD9A13A) // 警示
    static let amberText   = Color(hex: 0xECC57E)
    static let salmon      = Color(hex: 0xD98A6A) // 破壞性
    static let salmonText  = Color(hex: 0xEAB29A)
    static let offlineBlue = Color(hex: 0xB8C6DD) // 離線

    static var honeyGradient: LinearGradient {
        LinearGradient(colors: [honeyLight, honey], startPoint: .topLeading, endPoint: .bottomTrailing)
    }
}

File: app/wherebear_app_android/app/src/main/java/com/kerberosclaw/wherebear/ui/theme/BearTheme.kt
package com.kerberosclaw.wherebear.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color

/** 設計 tokens（深色底＋暖熊 accent）— 與 iOS BearTheme.swift 同一組色票 */
object BearTheme {
    val bg = Color(0xFF17130F)          // 主背景（暖近黑）
    val surface = Color(0xFF221B13)     // 卡片
    val surfaceHi = Color(0xFF262019)   // modal
    val sheet = Color(0xFF201913)       // bottom sheet
    val cream = Color(0xFFF4E3C8)       // 主文字
    val honey = Color(0xFFC98A4B)       // accent
    val honeyLight = Color(0xFFE8B878)  // accent 亮
    val ink = Color(0xFF2A1D10)         // accent 上的深字
    val pawInk = Color(0xFF3A2A18)      // 掌印
    val green = Color(0xFF7FAE63)       // 回報中
    val greenText = Color(0xFFCFE3BD)
    val amber = Color(0xFFD9A13A)       // 警示
    val amberText = Color(0xFFECC57E)
    val salmon = Color(0xFFD98A6A)      // 破壞性
    val salmonText = Color(0xFFEAB29A)
    val offlineBlue = Color(0xFFB8C6DD) // 離線

    val honeyGradient = Brush.linearGradient(listOf(honeyLight, honey))
}

private val WhereBearColors = darkColorScheme(
    primary = BearTheme.honeyLight,
    onPrimary = BearTheme.ink,
    secondary = BearTheme.honey,
    background = BearTheme.bg,
    onBackground = BearTheme.cream,
    surface = BearTheme.surface,
    onSurface = BearTheme.cream,
    surfaceVariant = BearTheme.surfaceHi,
    onSurfaceVariant = BearTheme.cream,
    error = BearTheme.salmon,
)

/** app 一律深色（對應 iOS 的 .preferredColorScheme(.dark)） */
@Composable
fun WhereBearTheme(content: @Composable () -> Unit) {
    @Suppress("UNUSED_EXPRESSION") isSystemInDarkTheme()
    MaterialTheme(colorScheme = WhereBearColors, content = content)
}
